app/src/main/java/com/humberto/tasky/agenda/presentation/AgendaViewModel.kt

```diff
@@ -94,22 +91,14 @@ class AgendaViewModel @Inject constructor(
         }
     }
 
-    private fun orderAgendaItems(agenda: Agenda): List<AgendaItemUi> {
-        val allItems = mutableListOf<AgendaItemUi>()
-        allItems.addAll(agenda.events.map { event -> event.toAgendaItemUi() })
-        allItems.addAll(agenda.tasks.map { task -> task.toAgendaItemUi() })
-        allItems.addAll(agenda.reminders.map { reminder -> reminder.toAgendaItemUi() })
-        return allItems.sortedBy { it.dateTime }
-    }
-
     private fun logout() {
         applicationScope.launch {
             _agendaState.update { state ->
                 state.copy(isLoggingOut = true)
             }
+            localAgendaDataSource.deleteAllAgenda()
             agendaRepository.logout()
             sessionManager.set(null)
```

Comment:
[Based on a comment from your previous PR 😊]
Here is an excellent example of wanting to suspend until the agenda items are deleted before logging the user out.